graphrag/index/flows/create_final_embeddings.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""All the steps to transform the text units."""
+
+import logging
+
+import pandas as pd
+from datashaper import (
+    VerbCallbacks,
+)
+
+from graphrag.index.cache import PipelineCache
+from graphrag.index.config.embeddings import (
+    community_full_content_embedding,
+    community_summary_embedding,
+    community_title_embedding,
+    document_raw_content_embedding,
+    entity_description_embedding,
+    entity_name_embedding,
+    relationship_description_embedding,
+    text_unit_text_embedding,
+)
+from graphrag.index.operations.embed_text import embed_text
+from graphrag.index.operations.snapshot import snapshot
+from graphrag.index.storage import PipelineStorage
+
+log = logging.getLogger(__name__)
+
+
+async def create_final_embeddings(
+    final_documents: pd.DataFrame,
+    final_relationships: pd.DataFrame,
+    final_text_units: pd.DataFrame,
+    final_entities: pd.DataFrame,
+    final_community_reports: pd.DataFrame,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    storage: PipelineStorage,
+    embedded_fields: set[str],
+    base_text_embed: dict | None = None,
+) -> pd.DataFrame:
+    """All the steps to generate all embeddings."""
+    documents_embeddings = final_documents.loc[:, ["id", "raw_content"]]
+    relationships_embeddings = final_relationships.loc[:, ["id", "description"]]
+    text_units_embeddings = final_text_units.loc[:, ["id", "text"]]
+
+    entities_embeddings = final_entities.loc[:, ["id", "name", "description"]]
+    entities_embeddings["name_description"] = (
+        entities_embeddings["name"] + ":" + entities_embeddings["description"]
+    )
+
+    community_reports_embeddings = final_community_reports.loc[
+        :, ["id", "full_content", "summary", "title"]
+    ]
```

Comment:
What do you think about doing this preprocessing at the verb level so that this function only takes what is needed to work?